src/ArrayOfVectorViews.jl

```diff
@@ -0,0 +1,85 @@
+struct ArrayOfVectorViews{T, N} <: AbstractArray{SubArray{T, 1, Vector{T}, Tuple{UnitRange{Int64}}, true}, N}
+    indices::Array{UnitRange{Int}, N}
```

Comment:
Are these always "dense" so that you will have

```
[1:2, 3:6, 7:15, ...]
```

in that case it is unnecessary to store the second number for all of these?

Reply:
Currently they are not guaranteed to be, that would require another sorting during compress, but could be a good optimization solving both the lower index storage requirement, and speeding up iteration somewhat. (If sizehint provided to the `ConstructionBuffer` is large enough, they are already "dense" after compressing)

Reply:
My guess is that it is worth it, especially for those cases where the "inner" arrays are short.

Reply:
Yes, but that feels like an optimization we can do later? 
From the outside, everything (including type parameters) will remain the same...

Reply:
Sure

Reply:
> (If sizehint provided to the ConstructionBuffer is large enough, they are already "dense" after compressing)

My comment above was wrong, that is only true if also started `add!`ed in that order, so some signficant sorted is required because normally this is not the case.